python/ray/experimental/workflow/api.py

```diff
@@ -56,4 +57,28 @@ def run(entry_workflow: Workflow, workflow_root_dir=None,
     return output
 
 
+# TODO(suquark): support recovery with ObjectRef inputs.
+
+
+def resume(workflow_id: str, workflow_root_dir=None) -> ray.ObjectRef:
+    """
+    Resume a workflow asynchronously. This workflow maybe fail previously.
+
+    Args:
+        workflow_id: The ID of the workflow. The ID is used to identify
+            the workflow.
+        workflow_root_dir: The path of an external storage used for
+            checkpointing.
+
+    Returns:
+        The execution result of the workflow, represented by Ray ObjectRef.
+    """
+    r = recovery.resume_workflow_job(workflow_id, workflow_root_dir)
+    if isinstance(r, ray.ObjectRef):
+        return r
+    # TODO(suquark): Currently this would override "steps/outputs.json".
```

Comment:
Just as a remainder, in this implementation we would override `steps/forward_outputs.json` to directly point to outputs. `steps/outputs.json` should not be modified for manual debugging (but modifying it does no harm).